docs/sources/developer/architecture/initialization.md

```diff
@@ -0,0 +1,85 @@
+# Initialization process
+
+The initialization process is pretty complex as it has to deal with a lot of components. Below is a step by step
+description of how it is initialized under the hood.
+
+Please refer to the components documentation for more details about each component.
+
+## Core package
+
+### 1. Unpatched Console
+
+The first component that is initialized is the unpatched console. This is needed because it is needed by the internal
+logger component. By default, the unpatched console is initialized with the global `console` object but it can be
+overwritten by the user if is provided in the config object.
+
+### 2. Internal Logger
+
+The internal logger is initialized immediate after the unpatched console but before anything else happens because it
+is needed to print out debug messages. The internal logger relies on the unpatched console and not on the `console`
+object because the latter can be overwritten by instrumentations and we don't want to send those debug messages as
+signals.
+
+### Isolation check
```

Comment:
If this happens before everything else, should we move it at the top?

Reply:
This doe snot. This comes after the unpatched console and the internal logger are initialized.